compiler/qsc_hir/src/hir.rs

```diff
@@ -78,6 +78,23 @@ impl From<NodeId> for usize {
     }
 }
 
+impl From<usize> for NodeId {
+    fn from(value: usize) -> Self {
+        NodeId(
+            value
+                .try_into()
+                .expect("Type Node ID does not fit into u32"),
```

Comment:
The message seems off, it's going from usize to u32, not NodeId to u32.

Reply:
Pushed updates. Deleted this code as it isn't used any more.